src/Compilers/Core/Portable/InternalUtilities/UnicodeCharacterUtilities.cs

```diff
@@ -109,6 +109,27 @@ public static bool IsValidIdentifier(string name)
             return true;
         }
 
+        /// <summary>
+        /// Returns true if every identifier in a namespace is valid using <see cref="IsValidIdentifier(string)"/>
+        /// </summary>
+        public static bool IsValidNamespace(string @namespace)
```

Comment:
I don't think this belongs in the compiler layer (the compiler doesn't work with a dotted name as a name).  Please move it to the editorfeatures layer.